logaut/backends/ltlf2dfa/core.py

```diff
@@ -124,3 +136,44 @@ def postprocess_output(output: str) -> str:
     if match is None:
         raise Exception("cannot find automaton description in MONA output.")
     return cast(Match, regex.search(output)).group(0)
+
+
+@singledispatch
+def get_atomic_propositions(formula: LTLfFormula) -> Set[str]:
+    """
+    Get the set of all atomic propositions in the LTLf formula.
+
+    :param formula: an LTLfFormula instance.
+    :return: a set of all atomic propositions.
+    """
+    raise NotImplementedError("Unsupported formula type")
+
+
+@get_atomic_propositions.register
+def _(formula: LTLfAtomic) -> Set[str]:
+    # If the formula is an atomic proposition, just return its symbol
+    return {formula.s}  # Assuming 's' is the symbol of the atomic proposition
+
+
+@get_atomic_propositions.register
+def _(formula: LTLfTrue) -> Set[str]:
+    # If the formula is a boolean constant, return empty
+    return set()
+
+
+@get_atomic_propositions.register
+def _(formula: LTLfFalse) -> Set[str]:
+    # If the formula is a boolean constant, return empty
+    return set()
+
+
+@get_atomic_propositions.register
+def _(formula: LTLfUnaryOperator) -> Set[str]:
+    # If the formula is a unary operator, recursively call for its child
+    return get_atomic_propositions(formula.f)
+
+
+@get_atomic_propositions.register
+def _(formula: LTLfBinaryOperator) -> Set[str]:
+    # If the formula is a binary operator, recursively call for its children and union the results
+    return set().union(*(get_atomic_propositions(f) for f in formula.formulas))
```

Comment:
Isn't this code already available within pylogics as `find_atoms()`? I don't remember exactly, but just recall that somewhere.